core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -180,9 +180,7 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
         try {
           val position = mmap.position()
 
-          /* Windows won't let us modify the file length while the file is mmapped :-( */
-          if (OperatingSystem.IS_WINDOWS)
-            safeForceUnmap()
+          safeForceUnmap()
```

Comment:
It may be safest to remap first and then unmap the old mmap. We perform some operations without acquiring a lock and calling `safeForceUnmap` seems a bit dangerous.